jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/messageHandler/MessageHandlerInterface.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019 Vantiq, Inc.
+ *
+ * All rights reserved.
+ * 
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.jmsSource.communication.messageHandler;
+
+import java.util.Map;
+import javax.jms.Message;
+import javax.jms.Session;
+
+import io.vantiq.extsrc.jmsSource.exceptions.UnsupportedJMSMessageTypeException;
+
+/**
+ * An interface used to handle incoming and outgoing JMS Messages for all Message Producers/Consumers/Listeners
+ */
+public interface MessageHandlerInterface {
+    
+    /**
+     * A helper function used to format the provided message according to whatever JMS Message Type was specified
+     * @param message           The message to be sent, either a string or a map
```

Comment:
Would it be easier for the case of String, to have a Map with a key of message in it with the string?  That seems more consistent and allows us to tighten up the interface.